### DISCUSSION TITLE
MATTER Source of TM12.5.0.4 arendst development in actual state ?

### DISCUSSION THREAD

**Discussion by HofeBY** [arendst/Tasmota#18907] [Support request]
Hi Tasmota Team
I installed the TasmotaMATTER ready Image 12.5.0.4 via Website to ESP32DevKitC from AZ-Delivery
After some configurations it looks like "TasmotaMATTER" device.
At the I2C Bus is a AHT21D wrong displayed as "VEML6070 UV ..."  . I know I have to unkomment in Source Code in file "tasmota_configurations_ESP32.h" and to build my own Version of TM12.5.0.4

To aktivate MATTER selve compiled Version in Source I add in file "user_config_override.h"

// -- Matter Protocol -freischalten mit SetOption151 1 -auch SO151 1 ---------
#undef USE_MATTER_DEVICE
#define USE_MATTER_DEVICE
// Enable all the crypto required by Matter
#undef  USE_BERRY_CRYPTO_EC_P256
#define USE_BERRY_CRYPTO_EC_P256
#undef  USE_BERRY_CRYPTO_HMAC_SHA256
#define USE_BERRY_CRYPTO_HMAC_SHA256
#undef  USE_BERRY_CRYPTO_HKDF_SHA256
#define USE_BERRY_CRYPTO_HKDF_SHA256
#undef  USE_BERRY_CRYPTO_AES_CCM
#define USE_BERRY_CRYPTO_AES_CCM
#undef  USE_BERRY_CRYPTO_AES_CTR
#define USE_BERRY_CRYPTO_AES_CTR
#undef  USE_BERRY_CRYPTO_PBKDF2_HMAC_SHA256
#define USE_BERRY_CRYPTO_PBKDF2_HMAC_SHA256
#undef  USE_BERRY_CRYPTO_SPAKE2P_MATTER
#define USE_BERRY_CRYPTO_SPAKE2P_MATTER

What do I get is shown in TasmotaClient "23ESP32". Tasmota Config Interface looks very diverent from Web Version below.

What do I have done wrong not getting the new MATTER Webversion ?

Device "20King3G" is a 3 Gang Switch Relais1 Shutter up, Relais 2 Shutter down; Relais 3 Light; and it has a BME280 Sensor via SDA1&SCL1 conneted to an ESP8285 CPU without the possibility to directly connecting to MATTER.

On Device "TasmotaMATTER" no values of "23ESP32 and 20King3G" are shown !

**Comment by s-hadinger** [comment]
To compile Matter on Tasmota, you just need `#define USE_MATTER_DEVICE`, all the crypto options are then automatically enabled. You also need to do `SetOption151 1` to enable Matter.

Without logs nor screenshots, it's impossible to help you.

**Comment by sfromis** [comment]
> At the I2C Bus is a AHT21D wrong displayed as "VEML6070 UV ..." . I know I have to unkomment in Source Code in file "tasmota_configurations_ESP32.h" and to build my own Version of TM12.5.0.4

You do not need your own build to handle such sensor overlap, disabling the VEML6070 driver can be done via the command `I2CDriver12 0`.

> On Device "TasmotaMATTER" no values of "23ESP32 and 20King3G" are shown !

Do not expect such values. In general, device configuration is done by finding the device in the [Tasmota Supported Devices Repository](https://templates.blakadder.com/), and following the procedure documented there. If not available there, you need some other way of working out the needed config details, and create a template to match.
